website/docs/reference/resource-properties/arguments.md

```diff
@@ -0,0 +1,116 @@
+---
+title: arguments
+sidebar_label: "arguments"
+id: arguments
+---
+
+import MacroArgsNote from '/snippets/_validate-macro-args.md';
+
+<File name='macros/<filename>.yml'>
+
+```yml
+
+version: 2
+
+macros:
+  - name: <macro name>
+    arguments:
+      - name: <arg name>
+        type: <string>
+        description: <markdown_string>
+
+```
+
+</File>
+
+## Definition
+
+The `arguments` property is used to define the parameters that a macro can accept. Each argument can have a `name`, `type`, and `description`.  You add `arguments` to a [macro property](/reference/macro-properties), which helps in documenting the macro and understanding what inputs it requires.
+
+You can validate your macro `arguments` using the [`validate_macro_args`](/reference/global-configs/behavior-changes#macro-argument-validation) flag:
+
+- If the flag is set to `False` (default), dbt will continue to permit any value for `type` and `name`.
+- If flag is set to `True` (opt-in), dbt will raise a warning if the argument names you've added in YAML don't match the argument names you have in your macro or if the argument types aren't valid according to the [supported types](/reference/global-configs/behavior-changes#supported-types).
+
+If no argument names are documented in YAML, dbt will infer them based on what you have in the macro and include them in the [manifest.json](/reference/artifacts/manifest-json) file.
+
+## type
+
+<VersionBlock lastVersion="1.9">
+
+The data type of your argument. This is only used for documentation purposes — there are no restrictions on the values you can use here.
+
+</VersionBlock>
+<VersionBlock firstVersion="1.10">
+
+The data type of your argument. Unless you use the [`validate_macro_args`](/reference/global-configs/behavior-changes#macro-argument-validation) flag, `type` is only used for documentation purposes — there are no restrictions on the values you can use here.
+
+</VersionBlock>
+
+<MacroArgsNote />
+
+<File name='macros/<filename>.yml'>
+
+```yml
+version: 2
+
+macros:
+  - name: <macro name>
+    arguments:
+      - name: <arg name>
+        type: <string>
+
+```
+
+</File>
+
+### Supported types for macro argument validation
```

Comment:
this might be just me but i feel this header can be shortened to just 'supported types' since it's nested under types and the intro text explains what its for. if you do change it, anything that links to this section will need updating